coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/StaticHiveFunctionRegistry.java

```diff
@@ -376,6 +376,8 @@ public RelDataType inferReturnType(SqlOperatorBinding opBinding) {
     createAddUserDefinedFunction("sort_array", ARG0, ARRAY);
     createAddUserDefinedFunction("extract_union", HiveReturnTypes.EXTRACT_UNION_FUNCTION_RETURN_STRATEGY,
         or(ANY, family(SqlTypeFamily.ANY, SqlTypeFamily.INTEGER)));
+    createAddUserDefinedFunction("coalesce_struct", HiveReturnTypes.COALESCE_STRUCT_FUNCTION_RETURN_STRATEGY,
+        or(ANY, family(SqlTypeFamily.ANY, SqlTypeFamily.INTEGER)));
```

Comment:
Given this method is registered, I think we'd better add the tests to verify the converted RelNode's type of SQLs like `SELECT coalesce_struct(a) from recursive_union`. That will give us more confidence than `CoalesceStructUtilityTest` only.
But yeah, it needs #192 to be checked in first.